GoogleForCreators/web-stories-wp
# Color Preset updates for beta

## Summary

Does the following:

- Temporarily removes Text Styles section from the presets and also makes necessary adjustments for not adding new text styles and only adding colors.
- Allows using presets for Page Background
- Disables presets with Opacity for Page + adds a Tooltip to reflect that it's disabled
- Adds transparent background pattern to all presets.
- Adds outline to color presets.


## Relevant Technical Choices

- Removes all the necessary parts from the code that are related to displaying/applying text styles, however, keeps the general logic in place for being able to add the style presets back easily post-beta.


## To-do

- [x] See what to do with Page Background

- [x] Tests for Page background



## User-facing changes

- Text style presets are not available anymore
- Presets will be added for color only now.
- Page background can use Presets
- Page background can not use presets with opacity.


## Testing Instructions
I
1. Add a text block
2. Change the background color, font style, the text color
3. Click on `+` in the color presets panel.
4. Verify that only the Color is added as a Preset and no Style Presets appear as before.

II
1. Add a Shape
2. Modify the Shape color
3. Click "Add to color presets" in the Color Picker
4. Verify the color was added

III
1. Add a few text elements
2. Select multiple text elements
3. Verify that the colors are applied to all elements when clicking on a Color Preset.

IV
1. Add Background Media
2. Change the Page background color by using a Color Preset
3. Remove the background Media
4. Ensure that the color preset was applied

V
1. Add some color presets, some of the with opacity
2. Select Page / Background
3. Verify that the color presets with opacity are grayed out and can't be applied.

Note: Shapes and Texts have different color preset, a color added via a Shape element will not appear under Text element presets.


---



Fixes #1837
Fixes #2186 
Fixes #2328 
Fixes #2308
